keyboards/kbd67/hotswap/keymaps/zunger/keymap.c

```diff
@@ -0,0 +1,163 @@
+ _______, H(03b6), H(03be), H(03c7), H(03c8), H(03b2), H(03bd), H(03bc), H(2272), H(2273), H(2208),                   _______, H(2191), H(2194), \
+ _______, _______, _______,                   _______,                            _______, _______,                   H(2190), H(2193), H(2192)),
+
+  /* Mathematical upper layer. Blank keys are TRNS.
+   * ,----------------------------------------------------------------.
+   * | `  | ₁| ₂ | ₃ | ₄ | ₅ | ₆ | ₇ | ₈ | ₉ | ₀ | ₋ | ₊ | ⊗ | ≈ |    |
+   * |----------------------------------------------------------------|
+   * |     | ∓ | Ω | Ε | Ρ | Τ | Θ | Υ | Ι | Ο | Π | ∄ | ∞ |     |    |
+   * |----------------------------------------------------------------|
+   * |       | Α | Σ | Δ | Φ | Γ | Η | ¬ | Κ | Λ | ∪ | ∩ |       |    |
+   * |----------------------------------------------------------------|
+   * |        | Ζ | Ξ | Χ | Ψ | Β | Ν | Μ | ⊆ | ⊇ | ∉ |      | ⇑ | ⇔  |
+   * |----------------------------------------------------------------|
+   * |    |    |    |                       |   |    ||||| ⇐ | ⇓ | ⇒  |
+   * `----------------------------------------------------------------'
+  */
+[_SHIFTMAGIC] = LAYOUT(
+ KC_GRV,  H(2081), H(2082), H(2083), H(2084), H(2085), H(2086), H(2087), H(2088), H(2089), H(2080), H(208b), H(208a), H(2297), H(2248), _______, \
+ _______, H(2213), H(03a9), H(0395), H(03a1), H(03a4), H(0398), H(03a5), H(0399), H(039f), H(03a0), H(2204), H(221e), _______,          _______, \
+ _______, H(0391), H(03a3), H(0394), H(03a6), H(0393), H(0397), H(00ac), H(039a), H(039b), H(222a), H(2229),          _______,          _______, \
+ _______, H(0396), H(039e), H(03a7), H(03a8), H(0392), H(039d), H(039c), H(2286), H(2287), H(2209),                   _______, H(21d1), H(21d4), \
+ _______, _______, _______,                   _______,                            _______, _______,                   H(21d0), H(21d3), H(21d2)),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  // We want magic to move us to layer _MAGIC, and shift+magic to move us to layer _SHIFTMAGIC. We
+  // can't do this with MO's because they get confused about the various orders in which keys can be
+  // pressed and released, so let's do it carefully and in code instead.
+  static bool shifted = false;
+  static bool magic = false;
+
+  if (keycode == KC_LSHIFT || keycode == KC_RSHIFT) {
+    shifted = record->event.pressed;
+  } else if (keycode == MAGIC) {
+    magic = record->event.pressed;
+  }
+
+  if (magic && shifted) {
+    layer_off(_MAGIC);
+    layer_on(_SHIFTMAGIC);
+  } else if (magic) {
+    layer_on(_MAGIC);
+    layer_off(_SHIFTMAGIC);
+  } else {
+    layer_off(_MAGIC);
+    layer_off(_SHIFTMAGIC);
+  }
+  return true;
+}
+
+void eeconfig_init_user(void) {
+  set_unicode_input_mode(UC_OSX);
+}
+
+void matrix_init_user(void) {
+  if (!eeconfig_is_enabled()) {
```

Comment:
This isn't actually needed. 

This is done a number of places in the core code, so this is actually redundant. 

For instance, this happens in audio and rgblight code.  It also happens in magic and bootmagic files, (bootmagic is added and called when that's enabled, or magic, when bootmagic isn't enabled).